### DISCUSSION TITLE
Is pastage of text when a configurable duration of time has elapsed supported?

### DISCUSSION THREAD

**Discussion by beedell-roke** [autokey/autokey#724] [Q&A] [answered]
I ask because I want to be able to duplicate a long authenticative key text into a window of a virtual-machine, but obviously the clipboard is unable to enable this. I consequently want to be able to select the input-box within the virtual-machine and wait some seconds until AutoKey duplicates it.

**Comment by Elliria** [comment]
It looks like you figured out `time.sleep()` on your own. You can also make that code a bit smaller, because `import time` isn't needed since AutoKey imports that for you by default.

**Reply by beedell-roke** [reply]
I really liked your previous response. I would be glad to accept it as the answer if you were to reinstate it.

I chose to retain `import time` because it allows the code to be utilized as python code if the necessary dependencies are installed.

**Reply by Elliria** [reply]
Ah, that makes sense. I do that, too, sometimes, since Python can go all over the place.

**Comment by Elliria** [comment]
Okay, I can do that.

**Comment by Elliria** [comment, marked as answer]
You can use `time.sleep()` to insert a delay into your code, specifying the amount of time in seconds. Below is a basic example that prints a line of text, waits 3 seconds, and prints another line of text:
```python
# Print the specified text:
keyboard.send_keys("This prints immediately.")

# Insert a three-second delay:
time.sleep(3)

# Print the specified text:
keyboard.send_keys('This prints after 3 seconds.')

```

**Comment by josephj11** [comment]
@BEEDELLROKEJULIANLOCKHART Glad this works for you.

Be aware that if you include any secrets as constants in your AutoKey scripts or phrases, they are stored as plain text in the body of the script or phrase. Depending on your application, this may be a security issue.

**Reply by beedell-roke** [reply]
@josephj11, might I be able to hook into native `python` libraries for this purpose? I expect that libraries that provide easy secure storage of authenticative keys via `python` already exist. I solely ask because I know not whether `autokey` restricts the libraries available.

**Reply by josephj11** [reply]
People import modules into scripts all the time. We don't restrict anything, but because Python can do almost anything, we don't really know in advance what will work or cause problems. Make a minimal test case with test data and see what happens. If you run into problems, come over to [Gitter](https://gitter.im/autokey/autokey) and ask for help (and show us how you did it!)

If I were doing this, I would get the encryption/decryption to work directly in the Python interpreter first before adding it into an AutoKey script. When you're outside of AutoKey, you can use all the features of the interpreter to debug your code or even use an IDE. Once you're inside AutoKey, it's much harder and slower to see what's going on and fix things.

I'm a Python beginner and don't do much with extra modules. There may be some issues with getting the modules you want to use into the path where Python will see them for importing. The folks on Gitter  can help with that if you run into it.

One extra for after it works: If you can write your encryption/decryption function without using any AutoKey API calls, you can save it in AutoKey's Modules directory and import it into any other script you write. If it needs our API, there's still a way to get it to work, but it's clunky. There's an enhancement request to get that fixed.
